src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -62,9 +62,17 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
 
   const scheduleAnimationCallback = config.scheduleAnimationCallback;
   const scheduleDeferredCallback = config.scheduleDeferredCallback;
+  const useSyncScheduling = config.useSyncScheduling;
 
-  // The default priority to use for updates.
-  let defaultPriority : PriorityLevel = LowPriority;
+  // The priority level to use when scheduling an update.
+  let priorityContext : (PriorityLevel | null) = null;
```

Comment:
You could stil keep the distinction in type system if you don't make it a part of the `PriorityContext` union and explicitly say `PriorityContext | NoPriorityContext`. Or something like it.